From e527707a34444b44647cb4915d3e4dae4fa961c1 Mon Sep 17 00:00:00 2001
From: LuoXiaoTan <lxt@rock-chips.com>
Date: Thu, 13 Apr 2017 20:24:51 -0700
Subject: [PATCH] arm64: dts: rk3399-sapphire: add tc358749x device

add tc358749x i2c dev & pin control for HDMIIn function
base on board (RK_EXCAVATOR_MAIN_V12_2016_11_11_FZB)

Change-Id: I5548a31d78e4d7cc9ecae0dc89f2ee2dcd0e75b4
Signed-off-by: LuoXiaoTan <lxt@rock-chips.com>
---
 .../rk3399-sapphire-excavator-edp-android-6.0.dts | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp-android-6.0.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp-android-6.0.dts
index 478bccbb8cc9..03afa71e3a45 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp-android-6.0.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp-android-6.0.dts
@@ -116,6 +116,21 @@
 		irq_gpio_number = <&gpio1 20 IRQ_TYPE_LEVEL_LOW>;
 		rst_gpio_number = <&gpio4 22 GPIO_ACTIVE_HIGH>;
 	};
+
+	tc358749x: tc358749x@0f {
+		compatible = "toshiba,tc358749x";
+		reg = <0x0f>;
+		power-gpios = <&gpio2 6 GPIO_ACTIVE_HIGH>;
+		power18-gpios = <&gpio2 9 GPIO_ACTIVE_HIGH>;
+		power33-gpios = <&gpio2 5 GPIO_ACTIVE_HIGH>;
+		csi-ctl-gpios = <&gpio2 10 GPIO_ACTIVE_HIGH>;
+		stanby-gpios = <&gpio2 8 GPIO_ACTIVE_HIGH>;
+		reset-gpios = <&gpio2 7 GPIO_ACTIVE_HIGH>;
+		int-gpios = <&gpio2 12 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&hdmiin_gpios>;
+		status = "okay";
+	};
 };
 
 &pwm0 {
-- 
2.35.3

